syntax = "proto3";
package board;

service Board{
    rpc AddAdvert (AddAdvertRequest) returns (AddAdvertReply) {}
    rpc GetCountOfAdverts (GetCountOfAdvertsRequest) returns (GetCountOfAdvertsReply) {}
    rpc GetAdverts (GetAdvertsRequest) returns (GetAdvertsReply) {}
    rpc ClearAdverts (ClearRequest) returns (ClearReply) {}
}

message AddAdvertRequest{
    string title = 1;
    string text = 2;
    string userName = 3;
    string userEmail = 4;
    string createdTime = 5;
}

message AddAdvertReply{
    string message = 1;
}


message GetCountOfAdvertsRequest{}

message GetCountOfAdvertsReply{
    int32 count = 1;
}

message GetAdvertsRequest{}

message GetAdvertsReply{
    repeated Advert adverts = 1;
}

message Advert{
    string title = 1;
    string text = 2;
    string userName = 3;
    string userEmail = 4;
    string createdTime = 5;
}

message ClearRequest{}

message ClearReply{
    int32 count = 1;
}
